aiutils/tftools/batch_normalizer.py

```diff
@@ -0,0 +1,94 @@
+"""A helper class for managing batch normalization state.
+
+This class is designed to simplify adding batch normalization
+(http://arxiv.org/pdf/1502.03167v3.pdf) to your model by
+managing the state variables associated with it.
+
+Important use note:  The function get_assigner() returns
+an op that must be executed to save the updated state.
+A suggested way to do this is to make execution of the
+model optimizer force it, e.g., by:
+
+  update_assignments = tf.group(bn1.get_assigner(),
```

Comment:
So, basically, what this is saying is that you should make it so that the optimizer has to run before the assigners run, then rather than running the optimizer, you should run the assigners, which will automatically run the optimizer.  I think the second group can probably be replaced with an identity.  But I'm not 100% sure (see below)

From the doc https://www.tensorflow.org/versions/r0.8/api_docs/python/framework.html#Graph.control_dependencies

```
# WRONG
def my_func(pred, tensor):
  t = tf.matmul(tensor, tensor)
  with tf.control_dependencies([pred]):
    # The matmul op is created outside the context, so no control
    # dependency will be added.
    return t
# RIGHT
def my_func(pred, tensor):
  with tf.control_dependencies([pred]):
    # The matmul op is created in the context, so a control dependency
    # will be added.
    return tf.matmul(tensor, tensor)
```